Build checks

markus2330 opened the pull request:
# Purpose

fix  #1966 

# Checklist

Check relevant points but please do not remove entries.
For docu fixes, spell checking, and similar nothing
needs to be checked.

- [X] commit messages are fine ("module: short statement" syntax and refer to issues)
- [ ] I added unit tests
- [ ] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)
- [ ] release notes are updated (doc/news/_preparation_next_release.md)

markus2330 commented:
@ingwinlu @sanssecours please review the PR. The PR currently fails on purpose to see how it looks like if the the release notes were not updated.

ingwinlu commented on scripts/jenkins/Jenkinsfile:
```diff
@@ -351,8 +352,17 @@ def build_todo() {
 }
 
+def build_check_release_notes() {
+    return withDockerEnv("check_release_notes", DOCKER_IMAGES.stretch) {
```
currently i use dashes instead of _ to separate words here. so it looks a bit inconsistent.

sanssecours commented on scripts/run_check_release_notes:
```diff
@@ -0,0 +1,15 @@
+if [ $? != "0" ]
+then
+	echo "The release notes were not extended correctly."
+	echo "Please make sure you added at least one line which says *(your name)* to"
```
I would replace “says” with “states”/“mentions” here or reformulate the sentence a little bit:

 > Please make sure you add at least one line describing you contribution followed by the text “*(your name)*”.

. There is also a minor typo at the end of the current sentence:

> …which says *(your name)* to**o.**

.

markus2330 commented:
> just not sure why you included the move of icheck in 4959aa9?

You mean why I moved it or why it was included in one commit together with other things?

1. I moved it because I think the check is done fast and its good to get quick feedback about it (e.g., see #1388).
2. It is in one commit because of laziness, it should have been two separate commits.

ingwinlu commented:
Another thing you might need to do is restrict the check to PR's only. We do not want to tag a commit in master as failed just because it did not modify release notes.

you can use the maybeStage for it and check if the branch isMaster():
```
maybeStage("blablaicheck", isMaster()) {
```

sanssecours commented on scripts/run_check_release_notes:
```diff
@@ -0,0 +1,21 @@
+		exit 0
+	fi
+	echo "The release notes were not extended correctly (with $L lines)."
+	echo "Please make sure you added at least one line which says *(your name)* to"
```
I would replace “says” with “states”/“mentions” here or reformulate the sentence a little bit:

 > Please make sure you add at least one line describing you contribution followed by the text “*(your name)*”.

.

sanssecours commented on scripts/run_check_release_notes:
```diff
@@ -0,0 +1,21 @@
+	echo "The release notes were not extended correctly (with $L lines)."
+	echo "Please make sure you added at least one line which says *(your name)* to"
+	echo "make sure that your contribution is correctly attributed in the release notes."
+	echo ""
```
The quotes after `echo` are useless, but I guess that is a matter of style preference. I also would prefer single quotes whenever possible, since using double quotes often has [unintended consequences](https://github.com/ElektraInitiative/libelektra/commit/aace3610003c184af7e6e9ec4d7f3f2f05b1d47f).

markus2330 replied:
I think they are quite useful if you start to write new lines to be echoed.

sanssecours commented on scripts/run_check_release_notes:
```diff
@@ -0,0 +1,21 @@
+git diff origin/master -- doc/news/_preparation_next_release.md | egrep "^\+.+\*\(.+\)\*"
+if [ $? != "0" ]
+then
+	L=git diff origin/master -- doc/news/_preparation_next_release.md | egrep "^\+" | wc -l
```
Please replace `L` with a more descriptive variable name such as `linesChanged`.

sanssecours replied:
I think you forgot to add [double back ticks or `$()`](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2013/5/pipeline/) here:

```
diff: origin/master: No such file or directory
0
scripts/run_check_release_notes: 7: [: Illegal number: 
The release notes were not extended correctly (with  lines).
```

.

ingwinlu commented:
Should this pr also change the pr template? Is there still a reason for the purpose field if the release notes need to be filled out?

markus2330 commented:
> maybeStage

I am not sure how to use maybeStage correctly. (I still do not know the return types of many functions.)

> Should this pr also change the pr template?

I considered adding something about the API compatibility, too. But we need to draw the line somewhere, changing APIs is not what new contributors try to do.

> Is there still a reason for the purpose field if the release notes need to be filled out?

New contributors won't read the whole documentation, so things that really need be done need to be pointed out at several spots. As I see it, the build server should only check if people forgot something.

ingwinlu commented:
> I am not sure how to use maybeStage correctly. (I still do not know the return types of many functions.)

Yeah withDockerEnv should not automagically provide a stage. I will have to rework that.

ingwinlu commented:
it is a non trivial change. I would rather change it in #2003.

ingwinlu commented:
I will add it here but the rebase for #2003 will not be fun :(

markus2330 commented:
We can also merge #2003 first or you move the commits of #2003 here.

markus2330 commented:
@ingwinlu seems like a7 is down :cry: 

Is this PR ready to merge even without trying it?

ingwinlu commented:
There is no rush from my side.

On 19 May 2018 at 21:33, markus2330 <notifications@github.com> wrote:

> @ingwinlu <https://github.com/ingwinlu> seems like a7 is down 😢
>
> Is this PR ready to merge even without trying it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ElektraInitiative/libelektra/pull/2013#issuecomment-390427384>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEOv-s6ODfp8rSJIp6gp8k9iKubKLbQhks5t0HN2gaJpZM4UFoMG>
> .
>

markus2330 commented:
Thanks for the help!
